lib/msal-node/src/client/ClientApplication.ts

```diff
@@ -109,4 +121,31 @@ export abstract class ClientApplication {
             },
         };
     }
+
+    /**
+     * Create authority instance. If authority not passed in request, default to authority set on the application
+     * object. If no authority set in application object, then default to common authority.
+     * @param authorityString
+     */
+    private async createAuthority(authorityString?: string): Promise<Authority> {
+        const authority: Authority = authorityString
+            ? AuthorityFactory.createInstance(authorityString,  this.appConfig.system!.networkClient!)
+            : this.authority;
+
+        await authority.resolveEndpointsAsync().catch(error => {
+            throw ClientAuthError.createEndpointDiscoveryIncompleteError(error);
+        });
+        return authority;
```

Comment:
If you are going to use `await` here, it is idiomatic to use a `try/catch` instead of `.catch`:

```js
try {
  await authority.resolveEndpointsAsync();
  return authority;
} catch (error) {
  throw ClientAuthError.createEndpointDiscoveryIncompleteError(error);
}
```

Reply:
+1

Reply:
Updated.